### PR TITLE
Migrating autosave from localstorage to indexDB

### LINKED ISSUE #1144: Use something other than localStorage for storing autosaves
localStorage is slow and synchronous, we use something faster for storing local changes. In #894, @SimonBiggs recommended using the dexie wrapper for indexedb: http://dexie.org/

### PR DESCRIPTION
**Pull request by iFlameing**
This pr fixes #1144.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,8 +45,10 @@
     "css-element-queries": "1.1.1",
     "date-fns": "2.0.0-alpha.26",
     "deep-equal": "1.0.1",
+    "dexie": "^2.0.4",
     "emotion": "9.2.12",
     "error-stack-parser": "2.0.2",
+    "fake-indexeddb": "^2.0.4",
```
Comment (wlach) [inline package.json]: This should be a devDependecy

**Comment by wlach** [comment]
> @wlach please close this PR if you wish, it's up to you since you're leading that section of work

I think I am ok to merge this now, as I haven't really started on the new autosave implementation yet.

File: test/globals-test.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (wlach) [inline test/globals-test.js]: what error are you getting if you remove this line?

Comment (iFlameing) [reply test/globals-test.js]: Nothing, I revert this change. I am thinking that I added  it when I don't install fake-indexdb

**Comment by iFlameing** [comment]
@wlach  I the test-javascript is failing due to test file doesn't giving any output for long time . I noticed that it is getting stuck at editor-mode-toolbar.js you can also see in this pr also test in stuck at the same location.
https://github.com/iodide-project/iodide/pull/1370

**Comment by wlach** [comment]
I think you are right, see #1374. I'll do one more pass through before merging later today.

**Pull request merged**